src/app/services/api-loading.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+/**
+ * @description This is an example implementation of the power of using an abstract service file. Allows for APIs to "register" with this service, so that it can monitor what service files are waiting on a response at a given time.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiLoadingService {
+  private apiList: ApiData[] = []; // Start with none, to be populated as each api is registered.
+  private apiListChanged: BehaviorSubject<ApiData[]> = new BehaviorSubject<ApiData[]>([]);
+
+  constructor() { }
+
+  /**
+   * @description - To be called in the constructor of the abstract api service file, so that each api service can register with this here service
+   * @param {string} apiName - The name of the service file to be logged in this service, to also be displayed in the api list page
```

Comment:
just a suggest which you dont have to do...but i found there isnt any use for the type inside the jsdoc since we are using typescript here...i think its one of those misunderstood things atlas set up in standards maybe?  im pretty sure they arent doing in the the nestjs code.   the types in jsdoc are used to help enforce types in vanilla js